cert_manager/client.py

```diff
@@ -143,6 +143,32 @@ def get(self, url, headers=None):
 
         return result
 
+    def get_paginated(self, url, headers=None, page_size=20):
+        """Submit a GET request to the provided URL with pagination iteration.
+
+        :param str url: A URL to query
+        :param dict headers: A dictionary with any extra headers to add to the request
+        :param int page_size: The number of results to be requested in each paginated request
+        :return obj: Yield a requests.Response object received as a response for each request
+        """
+        position = 0
+        result_json = []
+
+        while position == 0 or len(result_json) == page_size:
+            next_url = prepare_url(
+                url, {'position': position, 'size': page_size})
+            result = self.__session.get(next_url, headers=headers)
```

Comment:
Since we are using sessions already, it would probably make more sense to use the built-in function `Session.prepare_request`.

```python
req = Request('GET', url)
while position == 0 or len(result_json) == page_size:
    next_req = self.__session.prepare_request(req)
    next_req.params = {'position': position, 'size': page_size}
    result = self.__session.send(next_req)
```

I haven't tested the above code, but the documentation seems to suggest this is the way to go:

https://requests.readthedocs.io/en/master/user/advanced/#prepared-requests

Reply:
The `PreparedRequest` instantiated in your example does not expose `params` as a writable property, so in order to set query parameters one needs to use `prepare_url()` or the full `prepare()` method.